include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -3509,6 +3510,273 @@ namespace internal
 
       return mapping_info;
     }
+
+    /**
+     * This function provides information which DoF index is associated with
+     * a support point.
+     *
+     * @param[in] dof_handler DoFHandler providing DoF indices which are
+     * collected at support points.
+     * @param[in] dof_handler_sp DoFHandler with one component used to determine
+     * support point indices.
+     * @param[in] constraint AffineConstrains associated with @p dof_handler.
+     *   Only unconstrained DoFs are considered
+     * @return a tuple containing 0) local support point indices,
+     *   1) pointers to global DoF indices, and 2) global DoF indices.
+     */
+    template <int dim, int spacedim, typename Number>
+    std::tuple<std::vector<unsigned int>,
+               std::vector<unsigned int>,
+               std::vector<types::global_dof_index>>
+    support_point_indices_to_dof_indices(
+      const DoFHandler<dim, spacedim> &        dof_handler,
+      const DoFHandler<dim, spacedim> &        dof_handler_sp,
+      const dealii::AffineConstraints<Number> &constraint)
+    {
+      // in case a DG space of order 0 is provided, DoFs indices are always
+      // uniquely assigned to support points (they are always defined in the
+      // center of the element) and are never shared at vertices or faces.
+      Assert(
+        dof_handler_sp.get_fe().n_components() == 1 &&
+          (dof_handler_sp.get_fe().conforming_space ==
+             FiniteElementData<dim>::Conformity::H1 ||
+           dof_handler_sp.get_fe().degree == 0),
+        ExcMessage(
+          "dof_handler_sp has to be H1 conforming with one component or degree 0."));
+      Assert(&dof_handler.get_triangulation() ==
+               &dof_handler_sp.get_triangulation(),
+             ExcMessage("DoFhandlers need the same underlying triangulation."));
+      Assert(dof_handler.get_fe().degree == dof_handler_sp.get_fe().degree,
+             ExcMessage("DoFhandlers need the same degree."));
```

Comment:
Thank you for pointing out the missing `Asserts`. 
I added `Asserts` to ensure the function can only be used with `FE_DGQ` and `FE_Q`. I also added an `Assert` to ensure the same `n_dofs_per_cell` which is indeed needed for now (and has to be deleted later when the function is extended such that it can be used to identify DoF indices for the same support point in the context of multiple components).

[7b8ee1b](https://github.com/dealii/dealii/pull/15209/commits/7b8ee1bb535fcdadf8b8a4998e72dcf925708aee)